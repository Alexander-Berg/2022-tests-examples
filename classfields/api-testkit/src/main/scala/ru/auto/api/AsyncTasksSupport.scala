package ru.auto.api

import io.prometheus.client.Counter

trait AsyncTasksSupport {

  implicit val asyncTasksCounter: Counter = Counter
    .build("test_async_counter", "Async tasks counter for tests")
    .labelNames("name", "error")
    .create()
}
