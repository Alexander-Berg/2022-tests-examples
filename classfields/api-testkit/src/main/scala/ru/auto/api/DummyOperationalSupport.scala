package ru.auto.api

import java.util.function.Predicate
import io.prometheus.client.{Collector, CollectorRegistry}
import ru.yandex.vertis.ops.MetricsSupport
import ru.yandex.vertis.ops.codahale.CodahaleRegistry
import ru.yandex.vertis.ops.prometheus.PrometheusRegistry

trait DummyOperationalSupport {

  implicit val metricsSupport: MetricsSupport = new MetricsSupport {
    implicit override def codahaleRegistry: CodahaleRegistry = null

    implicit override def prometheusRegistry: PrometheusRegistry = prometheusRegistryDummy
  }

  implicit val prometheusRegistryDummy: PrometheusRegistry = new PrometheusRegistry {
    override def register[C <: Collector](c: C): C = c

    override def asCollectorRegistry(): CollectorRegistry = new CollectorRegistry()

    override def unregister(p: Predicate[Collector]): Unit = ()
  }
}
