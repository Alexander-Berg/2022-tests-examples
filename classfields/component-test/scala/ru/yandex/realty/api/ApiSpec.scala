package ru.yandex.realty.api

import akka.http.scaladsl.model.StatusCodes
import org.junit.runner.RunWith
import org.scalatestplus.junit.JUnitRunner
import ru.yandex.realty.AbstractBnbSearcherComponentTestSpec

import scala.language.implicitConversions

@RunWith(classOf[JUnitRunner])
class ApiSpec extends AbstractBnbSearcherComponentTestSpec {

  "GET /ping" should {

    "return OK" in {

      Get("/ping") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
        }

    }

  }

}
