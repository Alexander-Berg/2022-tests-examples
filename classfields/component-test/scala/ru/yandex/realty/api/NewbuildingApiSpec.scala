package ru.yandex.realty.api

import akka.http.scaladsl.model.StatusCodes
import org.junit.runner.RunWith
import org.scalatestplus.junit.JUnitRunner
import ru.yandex.realty.AbstractBnbSearcherComponentTestSpec
import ru.yandex.realty.api.ProtoResponse.NewbuildingContactsResponse
import ru.yandex.realty.componenttest.data.campaigns.Campaign_56576
import ru.yandex.realty.componenttest.data.sites.Site_57547
import ru.yandex.realty.model.phone.RealtyPhoneTags.MapsMobileTagName

import scala.language.implicitConversions

@RunWith(classOf[JUnitRunner])
class NewbuildingApiSpec extends AbstractBnbSearcherComponentTestSpec {

  import NewbuildingApiSpec._

  "GET /newbuilding/{newbuildingId}/contacts" should {

    "return newbuilding contacts" in {

      val newbuildingId = Site_57547.Id
      val expectedRedirectObjectId = newbuildingRedirectObjectId(newbuildingId)
      val phone = Campaign_56576.Proto.getRedirects(0).getSource

      stubTeleponyGetOrCreateOk(objectId = expectedRedirectObjectId, phone)
      stubTeleponyDeleteOk(objectId = expectedRedirectObjectId)

      Get(s"/newbuilding/$newbuildingId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[NewbuildingContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

    "return newbuilding contacts when tag is provided" in {

      val newbuildingId = Site_57547.Id
      val expectedRedirectObjectId = newbuildingRedirectObjectId(newbuildingId)

      stubTeleponyDeleteOk(objectId = expectedRedirectObjectId)

      Get(s"/newbuilding/$newbuildingId/contacts?tag=$MapsMobileTagName") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[NewbuildingContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

    s"return status=${StatusCodes.NotFound} when newbuilding does not exist" in {

      val nonExistingNewbuildingId = 10101010

      Get(s"/newbuilding/$nonExistingNewbuildingId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.NotFound)
        }

    }

    "return newbuilding contacts when failed to remove old redirects" in {

      val newbuildingId = Site_57547.Id
      val expectedRedirectObjectId = newbuildingRedirectObjectId(newbuildingId)

      stubTeleponyDeleteError(objectId = expectedRedirectObjectId)

      Get(s"/newbuilding/$newbuildingId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[NewbuildingContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

  }

}

object NewbuildingApiSpec {

  def newbuildingRedirectObjectId(newbuildingId: Long): String = {
    s"site_$newbuildingId"
  }

}
