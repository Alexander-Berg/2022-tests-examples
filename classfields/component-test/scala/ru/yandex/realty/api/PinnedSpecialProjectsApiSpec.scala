package ru.yandex.realty.api

import akka.http.scaladsl.model.StatusCodes
import org.junit.runner.RunWith
import org.scalatestplus.junit.JUnitRunner
import ru.yandex.realty.AbstractBnbSearcherComponentTestSpec
import ru.yandex.realty.componenttest.data.sites.Site_57547
import ru.yandex.realty.model.region.NodeRgid
import ru.yandex.realty.render.search.RenderableSearchResponse

@RunWith(classOf[JUnitRunner])
class PinnedSpecialProjectsApiSpec extends AbstractBnbSearcherComponentTestSpec {

  "GET /newbuildingSearch" should {

    "response with pinned sites in sbp region" in {
      Get(s"/newbuildingSearch?rgid=${NodeRgid.SPB_AND_LEN_OBLAST}") ~>
        route ~>
        check {
          status shouldBe StatusCodes.OK
          val response = entityAs[RenderableSearchResponse]
          response.result.pinnedItems.size shouldBe 1
          response.result.pinnedItems.head.id shouldBe Site_57547.Id
        }
    }

    "response with pinned sites in child geo of spb region" in {
      Get(s"/newbuildingSearch?rgid=${NodeRgid.SPB}&parkingType=UNDERGROUND") ~>
        route ~>
        check {
          status shouldBe StatusCodes.OK
          val response = entityAs[RenderableSearchResponse]
          response.result.pinnedItems.size shouldBe 1
          response.result.pinnedItems.head.id shouldBe Site_57547.Id
        }
    }

    "response with no pinned sites because of geo " in {
      Get(s"/newbuildingSearch?rgid=${NodeRgid.NIZHNY_NOVGOROD_OBLAST}") ~>
        route ~>
        check {
          status shouldBe StatusCodes.OK
          val response = entityAs[RenderableSearchResponse]
          response.result.pinnedItems shouldBe empty
        }
    }

    "response with no pinned sites because of filters " in {
      Get(s"/newbuildingSearch?rgid=${NodeRgid.SPB_AND_LEN_OBLAST}&parkingType=OPEN") ~>
        route ~>
        check {
          status shouldBe StatusCodes.OK
          val response = entityAs[RenderableSearchResponse]
          response.result.pinnedItems shouldBe empty
        }
    }
  }

}
