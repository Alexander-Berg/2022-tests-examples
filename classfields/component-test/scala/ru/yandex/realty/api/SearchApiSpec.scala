package ru.yandex.realty.api

import akka.http.scaladsl.model.StatusCodes
import org.junit.runner.RunWith
import org.scalatestplus.junit.JUnitRunner
import ru.yandex.realty.AbstractBnbSearcherComponentTestSpec
import ru.yandex.realty.componenttest.data.sites.Site_1754609
import ru.yandex.realty.componenttest.data.sites.Site_73030
import ru.yandex.realty.model.message.ExtDataSchema.SiteApartmentTypeNamespace.SiteApartmentType
import ru.yandex.realty.render.search.RenderableSearchResponse

@RunWith(classOf[JUnitRunner])
class SearchApiSpec extends AbstractBnbSearcherComponentTestSpec {

  "GET /newbuildingSearch" should {

    "return newbuilding by ID" in {
      val siteId = Site_73030.Id
      Get(s"/newbuildingSearch?siteId=$siteId") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[RenderableSearchResponse]
          val newbuilding = response.result.items.head
          newbuilding.id should be(siteId)
          newbuilding.buildingFeatures.apartmentType == SiteApartmentType.APARTMENTS_AND_FLATS
        }
    }

    "return sites with PLUS_4 rooms" in {

      Get("/newbuildingSearch?roomsTotal=PLUS_4") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[RenderableSearchResponse]
          response.result.items.map(_.id).exists(_ == Site_1754609.Id) should be(true)
        }

    }
  }

}
