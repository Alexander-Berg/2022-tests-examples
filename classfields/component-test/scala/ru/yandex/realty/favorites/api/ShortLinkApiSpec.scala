package ru.yandex.realty.favorites.api

import org.junit.runner.RunWith
import org.scalatestplus.junit.JUnitRunner
import ru.yandex.realty.favorites.{
  AbstractFavoritesComponentTestSpec,
  CreateOffersShortLinkRequest,
  CreateSitesShortLinkRequest,
  CreateVillagesShortLinkRequest,
  GetOffersRequest,
  GetSitesRequest,
  GetVillagesRequest
}
import ru.yandex.realty.proto.api.error.ErrorCode

import scala.collection.JavaConverters._

@RunWith(classOf[JUnitRunner])
class ShortLinkApiSpec extends AbstractFavoritesComponentTestSpec {

  "CreateOffersShortLink" should {

    "create valid link ID when several offers was received" in {
      // given
      val offerIds = Seq(
        "6937155746319343003",
        "3179879834876516998",
        "6053139209874256526"
      )

      // when
      val linkIdResponse = favoritesService.createOffersShortLink(
        CreateOffersShortLinkRequest
          .newBuilder()
          .addAllOfferIds(offerIds.asJava)
          .build()
      )

      // then
      linkIdResponse.hasResponse should be(true)

      val linkId = linkIdResponse.getResponse.getLinkId
      linkId.nonEmpty should be(true)

      // when
      val offerIdsResponse = favoritesService.getOffers(
        GetOffersRequest
          .newBuilder()
          .setLinkId(linkId)
          .build()
      )

      // then
      offerIdsResponse.hasResponse should be(true)

      val receivedOfferIds = offerIdsResponse.getResponse.getOfferIdsList.asScala
      receivedOfferIds should contain allElementsOf (offerIds)
    }

    "return error when received empty offer list" in {
      // given
      val emptyOfferIdsRequest =
        CreateOffersShortLinkRequest
          .newBuilder()
          .build()

      // when
      val response = favoritesService.createOffersShortLink(emptyOfferIdsRequest)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.EMPTY_FAVORITES_OFFER_IDS)
    }

    "return error when offers were not found by provided link" in {
      // given
      val nonExistentLinkId = shortLinkGenerator.generateShortLink()
      val request =
        GetOffersRequest
          .newBuilder()
          .setLinkId(nonExistentLinkId)
          .build()

      // when
      val response = favoritesService.getOffers(request)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.INVALID_FAVORITES_LINK_ID)
    }

    "return error when link has invalid format" in {
      // given
      val invalidLinkId = ""
      val request =
        GetOffersRequest
          .newBuilder()
          .setLinkId(invalidLinkId)
          .build()

      // when
      val response = favoritesService.getOffers(request)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.INVALID_FAVORITES_LINK_ID)
    }

  }

  "CreateSitesShortLink" should {

    "create valid link ID when several sites was received" in {
      // given
      val siteIds = Seq(
        57547L,
        1632667L,
        73030L
      )

      // when
      val linkIdResponse = favoritesService.createSitesShortLink(
        CreateSitesShortLinkRequest
          .newBuilder()
          .addAllSiteIds(siteIds.map(Long.box).asJava)
          .build()
      )

      // then
      linkIdResponse.hasResponse should be(true)

      val linkId = linkIdResponse.getResponse.getLinkId
      linkId.nonEmpty should be(true)

      // when
      val siteIdsResponse = favoritesService.getSites(
        GetSitesRequest
          .newBuilder()
          .setLinkId(linkId)
          .build()
      )

      // then
      siteIdsResponse.hasResponse should be(true)

      val receivedSiteIds = siteIdsResponse.getResponse.getSiteIdsList.asScala
      receivedSiteIds should contain allElementsOf (siteIds)
    }

    "return error when received empty site list" in {
      // given
      val emptySiteIdsRequest =
        CreateSitesShortLinkRequest
          .newBuilder()
          .build()

      // when
      val response = favoritesService.createSitesShortLink(emptySiteIdsRequest)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.EMPTY_FAVORITES_SITE_IDS)
    }

    "return error when sites were not found by provided link" in {
      // given
      val nonExistentLinkId = shortLinkGenerator.generateShortLink()
      val request =
        GetSitesRequest
          .newBuilder()
          .setLinkId(nonExistentLinkId)
          .build()

      // when
      val response = favoritesService.getSites(request)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.INVALID_FAVORITES_LINK_ID)
    }

    "return error when link has invalid format" in {
      // given
      val invalidLinkId = ""
      val request =
        GetSitesRequest
          .newBuilder()
          .setLinkId(invalidLinkId)
          .build()

      // when
      val response = favoritesService.getSites(request)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.INVALID_FAVORITES_LINK_ID)
    }

  }

  "CreateVillagesShortLink" should {

    "create valid link ID when several villages was received" in {
      // given
      val villageIds = Seq(
        57547L,
        1632667L,
        73030L
      )

      // when
      val linkIdResponse = favoritesService.createVillagesShortLink(
        CreateVillagesShortLinkRequest
          .newBuilder()
          .addAllVillageIds(villageIds.map(Long.box).asJava)
          .build()
      )

      // then
      linkIdResponse.hasResponse should be(true)

      val linkId = linkIdResponse.getResponse.getLinkId
      linkId.nonEmpty should be(true)

      // when
      val villageIdsResponse = favoritesService.getVillages(
        GetVillagesRequest
          .newBuilder()
          .setLinkId(linkId)
          .build()
      )

      // then
      villageIdsResponse.hasResponse should be(true)

      val receivedVillageIds = villageIdsResponse.getResponse.getVillageIdsList.asScala
      receivedVillageIds should contain allElementsOf (villageIds)
    }

    "return error when received empty village list" in {
      // given
      val emptyVillageIdsRequest =
        CreateVillagesShortLinkRequest
          .newBuilder()
          .build()

      // when
      val response = favoritesService.createVillagesShortLink(emptyVillageIdsRequest)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.EMPTY_FAVORITES_VILLAGE_IDS)
    }

    "return error when villages were not found by provided link" in {
      // given
      val nonExistentLinkId = shortLinkGenerator.generateShortLink()
      val request =
        GetVillagesRequest
          .newBuilder()
          .setLinkId(nonExistentLinkId)
          .build()

      // when
      val response = favoritesService.getVillages(request)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.INVALID_FAVORITES_LINK_ID)
    }

    "return error when link has invalid format" in {
      // given
      val invalidLinkId = "ABCDE0198475"
      val request =
        GetVillagesRequest
          .newBuilder()
          .setLinkId(invalidLinkId)
          .build()

      // when
      val response = favoritesService.getVillages(request)

      // then
      response.hasError should be(true)
      response.getError.getCode should be(ErrorCode.INVALID_FAVORITES_LINK_ID)
    }

  }

}
