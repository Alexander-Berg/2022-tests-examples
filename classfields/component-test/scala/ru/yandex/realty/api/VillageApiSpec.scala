package ru.yandex.realty.api

import akka.http.scaladsl.model.StatusCodes
import org.junit.runner.RunWith
import org.scalatestplus.junit.JUnitRunner
import ru.yandex.realty.AbstractBnbSearcherComponentTestSpec
import ru.yandex.realty.componenttest.data.villages.Village_1852045
import ru.yandex.realty.model.message.VillageCampaign
import ru.yandex.realty.model.phone.RealtyPhoneTags.MapsMobileTagName
import ru.yandex.realty.proto.village.api.VillageContactsResponse

import scala.language.implicitConversions

@RunWith(classOf[JUnitRunner])
class VillageApiSpec extends AbstractBnbSearcherComponentTestSpec {

  import VillageApiSpec._

  "GET /village/{villageId}/contacts" should {

    "return village contacts" in {

      val villageId = Village_1852045.Id
      val expectedRedirectObjectId = villageRedirectObjectId(Village_1852045.Campaign)
      val villagePhone = Village_1852045.DynamicInfo.getAuction
        .getSortedParticipants(0)
        .getSalesDepartment
        .getPhones(0)

      stubTeleponyGetOrCreateOk(objectId = expectedRedirectObjectId, source = villagePhone)
      stubTeleponyDeleteOk(objectId = expectedRedirectObjectId)

      Get(s"/village/$villageId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[VillageContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

    "return village contacts when tag is provided" in {

      val villageId = Village_1852045.Id
      val expectedRedirectObjectId = villageRedirectObjectId(Village_1852045.Campaign)

      stubTeleponyDeleteOk(objectId = expectedRedirectObjectId)

      Get(s"/village/$villageId/contacts?tag=$MapsMobileTagName") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[VillageContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

    s"return status=${StatusCodes.NotFound} when village does not exist" in {

      val nonExistingVillageId = 10101010

      Get(s"/village/$nonExistingVillageId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.NotFound)
        }

    }

    "return village contacts when failed to remove old redirects" in {

      val villageId = Village_1852045.Id
      val expectedRedirectObjectId = villageRedirectObjectId(Village_1852045.Campaign)

      stubTeleponyDeleteError(objectId = expectedRedirectObjectId)

      Get(s"/village/$villageId/contacts") ~>
        route ~>
        check {
          status should be(StatusCodes.OK)
          val response = entityAs[VillageContactsResponse]
          response.getResponse.getSalesDepartmentsCount === 1
        }

    }

  }

}

object VillageApiSpec {

  def villageRedirectObjectId(campaign: VillageCampaign): String = {
    campaign.getBilling.getKnownCampaign.getCampaign.getId
  }

}
