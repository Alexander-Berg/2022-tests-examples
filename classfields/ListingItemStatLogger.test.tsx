/**
 * @jest-environment @vertis/allure-report/build/circus/allure-jsdom-environment
 */

import React from 'react';
import { shallow } from 'enzyme';

jest.mock('auto-core/lib/event-log/statApi');
jest.mock('auto-core/react/lib/localStatData');
import { ContextBlock, ContextPage } from '@vertis/schema-registry/ts-types-snake/auto/api/stat_events';

import contextMock from 'autoru-frontend/mocks/contextMock';

import statApi from 'auto-core/lib/event-log/statApi';

import localStatData from 'auto-core/react/lib/localStatData';

import type { Offer } from 'auto-core/types/proto/auto/api/api_offer_model';

import type { Props } from './ListingItemStatLogger';
import ListingItemStatLogger from './ListingItemStatLogger';

const logImmediately = statApi.logImmediately as jest.MockedFunction<typeof statApi.logImmediately>;
const log = statApi.log as jest.MockedFunction<typeof statApi.log>;
const saveSearchClickInfo = localStatData.saveSearchClickInfo as jest.MockedFunction<typeof localStatData.saveSearchClickInfo>;

class TestListingItemStatLogger extends ListingItemStatLogger<Props> {
    render() {
        return null;
    }
}

beforeEach(() => {
    logImmediately.mockClear();
    log.mockClear();
});

it('на клик отправляет page, page_size и index правильно и записывает данные в ls', () => {
    const wrapper = shallow<TestListingItemStatLogger>(
        <TestListingItemStatLogger
            contextBlock={ ContextBlock.BLOCK_LISTING }
            contextPage={ ContextPage.PAGE_LISTING }
            index={ 4 }
            offer={{ category: 'cars' } as Offer}
            searchID="search_id"
            page={ 10 }
            pageSize={ 37 }
        />,
        { context: contextMock },
    );
    wrapper.instance().sendStatOnClick();
    expect(logImmediately).toHaveBeenCalledTimes(1);
    expect(logImmediately.mock.calls[0][0]).toMatchSnapshot();
    expect(saveSearchClickInfo)
        .toHaveBeenCalledWith('search_id', { category: 'cars' }, { block: 'BLOCK_LISTING', page: 'PAGE_LISTING', self_type: 'TYPE_SINGLE' });
});

it('должен отправить price_from и price_to в sendStatOnIntersectionChange', () => {
    const wrapper = shallow<TestListingItemStatLogger>(
        <TestListingItemStatLogger
            contextBlock={ ContextBlock.BLOCK_LISTING }
            contextPage={ ContextPage.PAGE_LISTING }
            index={ 4 }
            offer={{
                category: 'cars',
                groupping_info: {
                    price_from: { rur_price: 100 },
                    price_to: { rur_price: 200 },
                },
            } as Partial<Offer> as Offer}
            page={ 10 }
            pageSize={ 37 }
        />,
        { context: contextMock },
    );
    wrapper.instance().sendStatOnIntersectionChange(true);
    expect(log).toHaveBeenCalledTimes(1);
    expect(log.mock.calls[0][0].card_show_event?.price_from).toBe('100');
    expect(log.mock.calls[0][0].card_show_event?.price_to).toBe('200');
});

it('ничего не отправляет и не записывает в ls, если передан disableEventsLog', () => {
    const wrapper = shallow<TestListingItemStatLogger>(
        <TestListingItemStatLogger
            contextBlock={ ContextBlock.BLOCK_LISTING }
            contextPage={ ContextPage.PAGE_LISTING }
            index={ 4 }
            offer={{ category: 'cars' } as Offer}
            page={ 10 }
            pageSize={ 37 }
            disableEventsLog
        />,
        { context: contextMock },
    );
    wrapper.instance().sendStatOnClick();
    expect(logImmediately).not.toHaveBeenCalled();
    expect(saveSearchClickInfo).not.toHaveBeenCalled();
});
