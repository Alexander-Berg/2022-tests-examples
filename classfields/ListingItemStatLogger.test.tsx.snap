// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`на клик отправляет page, page_size и index правильно и записывает данные в ls 1`] = `
Object {
  "card_click_event": Object {
    "card_from": undefined,
    "card_id": "",
    "category": "CARS",
    "context_block": "BLOCK_LISTING",
    "context_page": "PAGE_LISTING",
    "search_position": 337,
    "search_query_id": "search_id",
    "section": "",
    "self_type": "TYPE_SINGLE",
  },
  "original_request_id": undefined,
}
`;
